Extend/Model/PaymentGateway.php

```diff
@@ -119,63 +109,114 @@ public function getRedirectUrls(Session $oSession, string $sShopUrl): Redirect
     /**
      * Executes the transaction through EE
      *
-     * @param float $fAmount
-     * @param Order $oOrder
+     * @param Basket $oBasket
+     * @param Order  $oOrder
      *
-     * @return FailureResponse|InteractionResponse|Response|SuccessResponse
+     * @return Transaction
      *
      * @throws Exception
      *
-     * @SuppressWarnings(PHPMD.Coverage)
      * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      * @SuppressWarnings(PHPMD.ElseExpression)
      * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
      */
-    public function makeTransaction(float $fAmount, Order $oOrder): Response
+    public function createTransaction($oBasket, $oOrder)
     {
         $sShopUrl = $this->getConfig()->getCurrentShopUrl();
+
         $oSession = $this->getSession();
-        $oBasket = $oSession->getBasket();
-        $sPaymentMethod = $oBasket->getPaymentId();
 
-        $oRedirect = self::getRedirectUrls($oSession, $sShopUrl);
-        $oPaymentMethod = Payment_Method_Factory::create($sPaymentMethod);
-        $oTransactionService = new TransactionService($oPaymentMethod->getConfig(), $this->_oLogger);
+        /**
+         * @var $oPayment Payment
+         */
+        $oPayment = oxNew(Payment::class);
+        $oPayment->load($oBasket->getPaymentId());
+
+        $sSid = $oSession->sid(true);
+        if ($sSid != '') {
+            $sSid = '&' . $sSid;
+        }
+
+        $oPaymentMethod = Payment_Method_Factory::create($oPayment->oxpayments__oxid->value);
 
         $oTransaction = $oPaymentMethod->getTransaction();
-        $oTransaction->setRedirect($oRedirect);
+        $oTransaction->setRedirect(self::getRedirectUrls($oSession, $sShopUrl, $sSid));
 
         $oShopconfig = $this->getConfig();
         $oCurrency = $oShopconfig->getActShopCurrencyObject();
-        $oTransaction->setAmount(new Amount($fAmount, $oCurrency->name));
-
-        $oBasket = $oSession->getBasket();
+        $oTransaction->setAmount(new Amount($oBasket->getPrice()->getBruttoPrice(), $oCurrency->name));
 
         $sOrderDetails = $oOrder->oxorder__oxremark->value;
```

Comment:
put the order details into an addOrderDetails and if its not empty you add it. You can also create a addMandatoryTransactionData() where you add notification url, redirects, amount, order detail.

Reply:
👍